### PR TITLE
build!: bump rust to 1.76 and add rust update script

### PR DESCRIPTION
**Pull request by lklimek**
## Issue being fixed or feature implemented

New Rust was released

## What was done?

* bumped version to 1.76 in github actions, rust-toolchain.toml, Dockerfile, Cargo.toml files and README.md
* added script that makes such updates easy

## How Has This Been Tested?

Run script locally + Github Actions

## Breaking Changes

New Rust version 1.76 is required

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: .github/actions/rust/action.yaml [hunk 1]
```diff
@@ -4,8 +4,8 @@ description: "Install dependencies"
 inputs:
   toolchain:
     description: Rust toolchain to use, stable / nightly / beta, or exact version
-    # The same as in /README.md
-    default: "stable"
+    # Update using scripts/update-rust-toolchain.sh
+    default: "1.76"
```
Comment (shumkov) [inline .github/actions/rust/action.yaml:8]: Isn't it's using toolchain file if not set?

File: .github/actions/rust/action.yaml [hunk 2]
```diff
@@ -17,7 +17,6 @@ inputs:
     description: Enable Rust cache
     required: false
     default: "true"
-
```
Comment (shumkov) [inline .github/actions/rust/action.yaml:20]: Why did you remove new lines? I see it more readable rather then on block of test

File: packages/dapi-grpc/Cargo.toml [hunk 3]
```diff
@@ -10,7 +10,8 @@ authors = [
     "Ivan Shumkov <shumkov@dash.org>",
 ]
 edition = "2021"
-rust-version = "1.73"
+# Update using scripts/update-rust-toolchain.sh
+rust-version = "1.76"
```
Comment (shumkov) [inline packages/dapi-grpc/Cargo.toml:14]: Why we set it here?

File: packages/rs-sdk/src/mock/wallet/minimal.rs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+use std::fmt::Display;
```
Comment (shumkov) [inline packages/rs-sdk/src/mock/wallet/minimal.rs:1]: Is this by mistake?

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/rs-sdk/src/mock/wallet/minimal.rs (hunk 4 above)
Comment (lklimek) [reply packages/rs-sdk/src/mock/wallet/minimal.rs:1]: yes, good point, missed this.

Will remove.

File: packages/dapi-grpc/Cargo.toml (hunk 3 above)
Comment (lklimek) [reply packages/dapi-grpc/Cargo.toml:14]: I don't understand the question.

The rust-version field is an optional key that tells cargo what version of the Rust language and compiler your package can be compiled with. 

As we only test on 1.76, we only support this version onwards.

Comment (shumkov) [reply packages/dapi-grpc/Cargo.toml:14]: Yeah, I've never seen such an option, and since it's a minimal value then it's all good.

File: .github/actions/rust/action.yaml (hunk 2 above)
Comment (lklimek) [reply .github/actions/rust/action.yaml:20]: That's what `yq` tool did that I use to replace version in yaml.

I can try to refactor to use multi-line `sed` if you wish.

Comment (shumkov) [reply .github/actions/rust/action.yaml:20]: Oh, you dynamically update it...

Comment (shumkov) [reply .github/actions/rust/action.yaml:20]: Maybe just read it from the toolchain file instead of hardcoding?

Suggestion from ChatGPT:
```
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
      # Checkout your repository

    - name: Extract Rust toolchain version
      id: rust_toolchain
      run: |
        TOOLCHAIN_VERSION=$(grep channel rust-toolchain.toml | awk '{print $3}' | tr -d '"')
        echo "TOOLCHAIN_VERSION=$TOOLCHAIN_VERSION" >> $GITHUB_ENV
        echo "::set-output name=version::$TOOLCHAIN_VERSION"
      # This command reads the rust-toolchain.toml, extracts the version, and sets it as an environment variable and step output

    # Use the extracted version in subsequent steps
    - name: Use Rust toolchain
      uses: actions-rs/toolchain@v1
      with:
        toolchain: ${{ steps.rust_toolchain.outputs.version }}
        override: true
```

I would use `$GITHUB_OUTPUT` though

**Review by shumkov — CHANGES_REQUESTED** [review]

**Comment by lklimek** [comment]
Superseded by #2070
